### ISSUE TITLE
[Web-Ext][Bug] Settings don't save, activities deleted

### ISSUE THREAD

**Issue by DougTownsend** [thomaschampagne/elevate#1093]
**Describe the bug**
I had the chrome extension open and added a target to the yearly progression tab. When I added it chrome just closed and when I opened it again the extension had updated. All of my activities are gone, which is not that big of a deal since I can just sync again, but the athlete settings do not save when I try to update them for the sync. At one point a JSON error popped up in the "Athlete settings" page when I was trying to update it, but I can't reproduce that error message now and did not save the message. In my chrome "local extension settings" folder there is an "ldb" file that was updated the last time I synced, so I assume that has the activities in it, but since it is not in the JSON format expected by the "restore profile" button I can't use it.

Changes made to the global settings and zone settings do not save as well. The only setting that seems to save between closing and reopening is the dark mode vs light mode.

**To Reproduce**
Change any setting, close the browser and reopen.

**Environment**
 - OS: Linux
 - Browser name and version: Chrome v99
 - Web extension version 7.0.0-beta.2

**Comment by thomaschampagne** [comment]
Indeed some others users have experienced the issue too while upgrading to v7. It wasn't supposed to happen ofc...

If you have some issues to save settings, I suggest you to uninstall/install back the extension from here: https://chrome.google.com/webstore/detail/elevate-for-strava/dhiaggccakkgdfcadnklkbljcgicpckn

Sorry for that !

**Comment by A1Dan75** [comment]
I imported a backup in the end and resynced recent activities.  Worked fine, but now they've all gone again, and it's insiting on syncing all from scratch... not sure if it's to do with the new update (colours have changed?)

**Comment by thomaschampagne** [comment]
@A1Dan75 You experienced also the "activities wipe" during the update to v7. You have to re-sync all your activities unfortunatly. 

I'm still trying to understand what happened :/ I'm sorry

**Comment by A1Dan75** [comment]
Looks like it, it says my last backup isn't compatible.

Managed 400 then died, guess I'll have to do it over a few days.

**Comment by dolohow** [comment]
Happened to me as well, however I had backup of my whole chrome directory, so I reverted it.  If you need some help with debugging let me know, however I restrain myself from running it again, since the update will overwrite again my data.

**Comment by A1Dan75** [comment]
> Happened to me as well, however I had backup of my whole chrome directory, so I reverted it. If you need some help with debugging let me know, however I restrain myself from running it again, since the update will overwrite again my data.

So have you reverted to V6?  
Will it not automatically try and upgrade itself to 7 again?

**Comment by dolohow** [comment]
It will, therefore I can only run it without Internet.  My main browser is Firefox, therefore I am waiting for a fix to open chrome again and have _elevate_ update being less buggy with new release.

**Comment by dolohow** [comment]
Apparently you can disable automatic updates
https://stackoverflow.com/questions/27657617/how-to-disable-google-chrome-extension-autoupdate

I just did

**Comment by sugarfreejaz** [comment]
all activities gone for me too....can my strava archive be imported rather than waiting days for re-sync?

**Comment by sugarfreejaz** [comment]
I see this screen.  It syncs some activities, get the message to wait 4-5hours and the tab does not close itself and does not continue where it left off on my next attempt to continue the sync.  I have re-synced the same initial batch 3 times now.  Each sync starts off at zero activities.

![Screen Shot 2022-04-13 at 7 31 48 am](https://user-images.githubusercontent.com/103499808/163058157-d8054f05-3a33-403e-b415-d3326388af63.jpg)

**Comment by sugarfreejaz** [comment]
Answered my own question from earlier, but have no idea where to find this "File Connector".   Anyone?  Regular peice-meal sync over days is not working for me, this seems like the better option if I can get some clarity.
![Screen Shot 2022-04-13 at 7 41 39 am](https://user-images.githubusercontent.com/103499808/163059392-4b79d844-2a24-4304-854c-8a7e1facbf37.jpg)

**Comment by thomaschampagne** [comment]
> I see this screen. It syncs some activities, get the message to wait 4-5hours and the tab does not close itself and does not continue where it left off on my next attempt to continue the sync. I have re-synced the same initial batch 3 times now. Each sync starts off at zero activities.

That's normal. Just let it go further.

**Comment by thomaschampagne** [comment]
> Answered my own question from earlier, but have no idea where to find this "File Connector". Anyone? Regular peice-meal sync over days is not working for me, this seems like the better option if I can get some clarity.

The File connector is just above the strava one:
![image](https://user-images.githubusercontent.com/151973/163066331-ae286823-29cf-44ba-bfda-29de2048d237.png)

**Comment by funk44** [comment]
What a nightmare
1. Opened elevate and got asked to recalculate
2. Clicked it without realising what this meant - all activities dropped
3. Sync starts again, takes basically an entire day
4. Reopen, add some more details like age and practice level
5. Get prompted to calculate again which I absentmindedly click without releasing this is what caused my activities to be dropped and resynced in the first place
6. Re-sync starts again
7. In frustration, come onto this thread and see the recommendation to uninstall/reinstall, which I do
8. Reopen elevate and realise my entire FTP history is now gone which I put together from TrainerRoad, which I no longer have
9. Load intervals.icu and click the donate button

**Comment by Truck5067** [comment]
I cannot sync All Activities. I think the problem was due to a system reboot due to critical Dirty Pipe patch while activities were being imported
Tried several times to resync all activities -all attempts failed to sync any activities- looks like Strava have banned me - cannot sync any activities.

.The I  uninstalled and reinstalled extension - logged out of Strava for about an hour then trued to sync all activities and got 0 imported band 4-5 hour wait.

OS Linux
Extensuon 7.0.0-beta2
Chromium Version 100.0.4896.88 (Official Build) snap (64-bit)
Kernel 5.13.0-39-generic

**Comment by thomaschampagne** [comment]
> Does it continue the sync automatically?

Yes. Just let it go. It scan existing activites at the beginning then sync the new ones until the end.

**Comment by Truck5067** [comment]
Thomas - I am having trouble syncing all activities (about 1800) . U have disabled suspend on my laptop and after 2 days it has done 580.

Trying the MacOS version on an iMac it has done 923 in the same period - I have disabled power saving / suspend on the iMac.

I had little trouble with version 6 but can only remember doing 2 full sync of activities.

I think you app is very popular and you are over worked. I will donate next week.
